json/src/main/java/org/openjdk/skara/json/JSONArray.java

```diff
@@ -39,6 +39,24 @@ public JSONArray(JSONValue[] array) {
         }
     }
 
+    private void append(JSONValue value) {
```

Comment:
Could you rename this to `concat` and explicitly take a `JSONArray` instead as parameter? I'm fine with having `append`, but `append` should be just `append` (that is, append a single item to the array).

Reply:
To me there's no deeper semantic difference between "append" and "concat". To be precise, this method do a flattening append/concat, and should probably be named something like addAndFlatten to be completely unambiguous. But that kind of naming did not seem to fit in with the rest of the design here.

If you insist that "concat" means "flatten while adding" but "append" does not, sure, I can rename it.

Reply:
Ah, sorry, now I see what you mean. You want a concat(JSONArray a) instead. Well, it might be good to have, but that does not really solve my problem, does it? :-)

Reply:
Well, if you have `append(JSONValue v)` and `concat(JSONArray a)` then code using such an API can easily have the code:

```java
void appendOrConcat(JSONArray a, JSONValue v) {
    if (v.isArray()) {
        a.concat(v.asArray());
    } else {
        a.append(v);
    }
}
```

I'm fine with having `appendOrConcat` in `JSONArray` as well, but then I want it named `appendOrConcat`, not `append`. I agree that `append` is shorter, but it is too misleading of a name IMHO for a method that really does `appendOrConcat`.

Reply:
I still don't get it. Your example code would not recursively inline arrays; it will only work for a single nested layer. I can't see how that would be helpful to amything but the most specific situations. What is it you want to achieve?